src/interactive-window/commands/commandRegistry.web.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable, optional } from 'inversify';
+import { IExtensionSingleActivationService } from '../../platform/activation/types';
+import { IWorkspaceService } from '../../platform/common/application/types';
+import { IDisposable, IDisposableRegistry } from '../../platform/common/types';
+import { IExportCommands } from '../types';
+
+@injectable()
+export class CommandRegistry implements IDisposable, IExtensionSingleActivationService {
+    constructor(
+        @inject(IDisposableRegistry) private readonly disposables: IDisposableRegistry,
+        @inject(IExportCommands) @optional() private readonly exportCommand: IExportCommands | undefined,
+        @inject(IWorkspaceService) private readonly workspace: IWorkspaceService
+    ) {
+        if (!this.workspace.isTrusted) {
+            this.workspace.onDidGrantWorkspaceTrust(this.registerCommandsIfTrusted, this, this.disposables);
+        }
+    }
+
+    async activate(): Promise<void> {
+        this.registerCommandsIfTrusted();
+    }
+
+    private registerCommandsIfTrusted() {
+        if (!this.workspace.isTrusted) {
+            return;
+        }
+        this.exportCommand?.register();
+    }
+
+    dispose(): void {
```

Comment:
Not required, the `this.disposables` is actually the global registry, that will get cleared when vscode shutdowns or extension is unloaded.

Reply:
We did the same in https://github.com/microsoft/vscode-jupyter/blob/8f7f38e2bc664836af236e8e2ad9f1b85130c3c9/src/interactive-window/commands/commandRegistry.node.ts#L108 . I wonder if we need local `disposables` then.

Reply:
There's no need, you can just pass that in here.
We tend to use local variables (to track the list of disposables)  for classes that get created and later disposed (e.g. class created via a factory that later gets disposed, e.g. creating kernels and the like). 
This is a singleton and the lifetime of this class is tied to the lifetime of the vscode session, hence there's no need to dispose anything explicitly here.